nexusmods-app: 0.10.2 -> 0.11.3

MattSturgeon opened the pull request:
- Bump version: 0.10.2 -> 0.11.3
  - Upstream changelogs:
    [0.11.1](https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.11.1)
    [0.11.3](https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.11.3)
- [Previous update PR](https://github.com/NixOS/nixpkgs/pull/403082)

Fixes #410442

## Notes for end-users

> [!NOTE]
> Since 0.8.2, all games except Stardew Valley have been moved behind the "Unsupported Games" flag in Settings. If you were modding any other games in a previous release, you will need to enable this option to continue managing your mods.
>
> Since 0.10.2, Cyberpunk 2077 is also "supported".

> [!TIP]
> When upgrading from version 0.7.0 or newer, it _should_ be safe to upgrade **without** resetting/uninstalling.

> [!CAUTION]
> When upgrading from a **version older than 0.7.0**, you will need a fresh start.
> If you have configuration leftover from an old version, the updated app will crash.
>
> See upstream's guide on [how to uninstall the app](https://nexus-mods.github.io/NexusMods.App/users/Uninstall).

<details><summary>Resetting tips</summary>
<p>

You can reset all modded games and wipe your nexusmods-app config using:
```
NexusMods.App uninstall-app
```
- This should be done **before** updating the app.
- **Be careful**: There's no way to recover a wiped config!

If you've run the appimage version from upstream, or a pre-0.6 nixpkgs version, you may also need to remove old desktop entries to avoid other issues:
```
rm ~/.local/share/applications/com.nexusmods.app.desktop
```

</p>
</details>

> [!CAUTION]
> Known issues are listed in the [changelog](https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.11.3).

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [ ] Tested, as applicable:
  - [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Updated existing a release notes entry
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MattSturgeon committed:
nexusmods-app: 0.10.2 -> 0.11.3

https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.11.1
https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.11.3

MattSturgeon commented:
Should I backport to 25.05? If so, I'll bump the release note too.

Given NexusMods.App is still unstable and in development, I'm _hesitant_ to backport updates.

corngood commented:
> Given NexusMods.App is still unstable and in development, I'm hesitant to backport updates.

To me this could be an argument in favour of backporting.  Is 0.10.2 really going to be usable long-term?  It doesn't look like they've ever pushed a patch update after a minor update, so I'd treat 0.10.2 as unsupported.

I don't really have a strong opinion either way, but we should either cut off 25.05 now, or plan on backporting everything < 2.0.

MattSturgeon commented:
> Is 0.10.2 really going to be usable long-term? It doesn't look like they've ever pushed a patch update after a minor update, so I'd treat 0.10.2 as unsupported.

Usable? Probably yes. Supported? Probably not.

The main reason we didn't backport to 24.11 was the same reason we have a release note in 25.05; breaking changes to app-state requiring the user to "uninstall" manually.

That side of the app looks to have stabilised to the point where it will usually import old app-state without issue; so we no longer need to avoid backports due to known upgrade issues.

If we want to maintain having a "supported" version on the stable channel, we should backport all updates. I believe only the latest version is ever considered "supported".

If that is what we want, I'll follow your earlier advice and make the release note wording less version-specific.

corngood commented:
> If that is what we want

I don't think there's a clear right answer, so I'd say it's your call as maintainer.  

Does the app depend on any online services that may not be supported long-term?  That's the only other potential issue that comes to mind.

I guess an argument _against_ backporting is that the user can make the choice themselves by switching channels.

MattSturgeon committed:
doc/rl-2505: reword nexusmods-app entry

We now plan to backport updates, so the release note shouldn't specify
exact versions.

Reworded to improve clarity.

Co-authored-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

MattSturgeon commented:
I've updated the release note to not reference specific versions. I've also bumped the update to 0.11.3.

l0b0 reviewed (approved):
LGTM, thanks!

MattSturgeon committed:
doc/rl-2505: move nexusmods-app entry back to "incompatibilities"

It was inadvertently moved when additional sections were added in #393738

MattSturgeon commented:
I spotted that the release note was accidentally moved out of the "incompatibilities" section to "other changes" (https://github.com/NixOS/nixpkgs/pull/393738/files#r2105545829), so I've moved it back to "incompatibilities" in an extra commit on this PR.

MattSturgeon commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 409206`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>nixpkgs-manual</li>
  </ul>
</details>

GaetanLepage reviewed (approved):
LGTM

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 409206`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>nixpkgs-manual</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>nixpkgs-manual</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
  </ul>
</details>

JohnRTitor reviewed (commented):
https://github.com/NixOS/nixpkgs/pull/409206#discussion_r2105773211

MattSturgeon commented on doc/release-notes/rl-2505.section.md:
```diff
@@ -247,6 +247,10 @@
 - `strawberry` has been updated to 1.2, which drops support for the VLC backend and Qt 5. The `strawberry-qt5` package
   and `withGstreamer`/`withVlc` override options have been removed due to this.
 
+- `nexusmods-app` has been upgraded from a version with an incompatible app state format. Before upgrading, you **must reset all app state** (mods, games, settings, etc). NexusMods.App will crash if any state from a version older than 0.7.0 is still present.
+  - Typically, you can can reset to a clean state by running `NexusMods.App uninstall-app`. See Nexus Mod's [how to uninstall the app](https://nexus-mods.github.io/NexusMods.App/users/Uninstall) documentation for more detail and alternative methods.
+  - This should not be necessary going forward, because loading app state from 0.7.0 or newer is now supported. This is documented in the [0.7.1 changelog](https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.7.1).
```
Thanks, that's definitely an improvement!

However one of the goals with updating the entry was to avoid specifying exact version. Doing so means updating the entry every time we backport, which @corngood suggested isn't ideal.

Would something like this work:


```suggestion
- `nexusmods-app` has been upgraded from version 0.6.3. If you were running a version older than 0.7.0, then before upgrading, you **must reset all app state** (mods, games, settings, etc). Otherwise, NexusMods.App will crash due to app state files incompatibility.
  - Typically, you can can reset to a clean state by running `NexusMods.App uninstall-app`. See Nexus Mod's [how to uninstall the app](https://nexus-mods.github.io/NexusMods.App/users/Uninstall) documentation for more detail and alternative methods.
  - This should not be necessary going forward, because loading app state from 0.7.0 or newer is now supported. This is documented in the [0.7.1 changelog](https://github.com/Nexus-Mods/NexusMods.App/releases/tag/v0.7.1).
```

Or do you not see an issue with bumping the release note each time we update the package?

JohnRTitor replied:
Your suggestion LGTM.
